### PR TITLE
【HigherOrderAD】Define sigmoid as primitive operator and lowering silu into it

### PR DESCRIPTION
**Pull request by cxxly**
### PR Category

CINN

### PR Types

Improvements

### Description

Pcard-66975
**问题描述**
遇到 1/1+e^x 运算， 导数表达式为 e^x/(1+e^x)，x交大时，逐步运算，会出现 0 * inf，从而出NaN
类似问题：https://jax.readthedocs.io/en/latest/notebooks/Custom_derivative_rules_for_Python_code.html#numerical-stability

silu在一阶拆解时，包含1/1+e^-x ，从而二阶运算时可能导致出NaN
5297ea171358364d18a222a33b571308

### CODE REVIEW

**Review by cyber-pioneer — DISMISSED** [review]

File: paddle/fluid/prim/api/composite_backward/composite_double_backward_api.h
Comment (HydrogenSulfate) [inline paddle/fluid/prim/api/composite_backward/composite_double_backward_api.h:1]: LGTM

**Review by HydrogenSulfate — DISMISSED** [review]

**Review by cyber-pioneer — APPROVED** [review]

**Review by zyfncg — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM
